### PR TITLE
feat: mf-5790 send post on farcaster

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-5790

### CODE REVIEW

File: src/providers/warpcast/SocialMedia.ts [hunk 1]
```diff
@@ -48,6 +47,10 @@ import { createSessionByCustodyWallet } from '@/providers/warpcast/createSession
 import { createSessionByGrantPermission } from '@/providers/warpcast/createSessionByGrantPermission.js';
 import { WarpcastSession } from '@/providers/warpcast/Session.js';
 
+function toFid(profileId: string) {
```
Comment (guanbinrui) [inline src/providers/warpcast/SocialMedia.ts:50]: Move to `@/helpers/toFid.ts`

File: src/providers/hubble/proto/message.ts [hunk 2]
```diff
@@ -333,6 +318,8 @@ export interface Message {
     dataBytes?: Uint8Array | undefined;
 }
 
+const appla = '10';
```
Comment (guanbinrui) [inline src/providers/hubble/proto/message.ts:321]: ?

File: src/components/Compose/ComposeSend/index.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    const charsLength = chars.length;
+
+    const disabled = (charsLength === 0 || charsLength > 280) && images.length === 0 && !video;
+    const sendLens = useSendLens();
+    const sendFarcaster = useSendFarcaster();
+
+    const isLens = currentProfile?.source === SocialPlatform.Lens;
+    const [{ loading }, handleSend] = useAsyncFn(async () => {
+        if (isLens) {
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend/index.tsx:27]: We should support to send a post on both platform.

File: src/components/Compose/ComposeSend/index.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                <div className=" flex items-center gap-[10px] whitespace-nowrap text-[15px] text-main">
+                    <CountdownCircle count={charsLength} width={24} height={24} className="flex-shrink-0" />
+                    <span className={classNames(disabled ? ' text-danger' : '')}>{charsLength} / 280</span>
+                </div>
+            ) : null}
+
+            <button
+                className={classNames(
+                    ' flex h-10 w-[120px] items-center justify-center gap-1 rounded-full bg-black text-[15px] font-bold text-white dark:bg-white dark:text-black',
+                    disabled ? ' cursor-no-drop opacity-50' : '',
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend/index.tsx:45]: `disabled:cursor-no-drop disabled:opacity-50`

File: src/components/Compose/ComposeSend/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+export default function ComposeSend() {
+    const { chars, images, video } = useComposeStateStore();
+
+    const charsLength = chars.length;
+    const disabled = (charsLength === 0 || charsLength > 280) && images.length === 0 && !video;
+    const sendLens = useSendLens();
+    const sendFarcaster = useSendFarcaster();
+
+    const [{ loading }, handleSend] = useAsyncFn(async () => {
+        await Promise.allSettled([sendLens(), sendFarcaster()]);
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend/index.tsx:21]:
```ts
const { source } = useComposeStateStore()

if (!source) {
    // post on both platforms
} else if (isLens) {
    // post to lens
} else if (isFarcaster) {
    // post to farcaster
} else {
    safeUnreachable()
}
```

File: src/components/Compose/ComposeSend/useSendFarcaster.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { useFarcasterStateStore } from '@/store/useFarcasterStore.js';
+
+export function useSendFarcaster() {
+    const { type, chars: content } = useComposeStateStore();
+    const enqueueSnackbar = useCustomSnackbar();
+    const currentProfile = useFarcasterStateStore.use.currentProfile?.();
+
+    return useCallback(async () => {
+        if (!currentProfile?.profileId) return;
+        if (type === 'compose') {
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend/useSendFarcaster.ts:18]: Need implementation for `replay`.

File: src/components/Compose/PostByItem.tsx [hunk 7]
```diff
@@ -52,71 +51,64 @@ export function PostByItem({ source }: PostByItemProps) {
...
-                        if (source === SocialPlatform.Farcaster && images.length > 2) {
-                            enqueueSnackbar(t`Select failed: More than 2 images`, {
-                                variant: 'error',
-                            });
-                            return;
-                        }
+        <div className=" flex h-[40px] items-center justify-between border-b border-secondaryLine last:border-none">
+            <div className=" flex items-center gap-2">
+                <SourceIcon size={24} source={source} />
+                <span className={classNames(' font-bold text-main')}>{resolveSourceName(source)}</span>
```
Comment (guanbinrui) [inline src/components/Compose/PostByItem.tsx:90]:
```suggestion
                <span className=" font-bold text-main">{resolveSourceName(source)}</span>
```

File: src/components/Compose/PostByItem.tsx [hunk 8]
```diff
@@ -52,71 +51,64 @@ export function PostByItem({ source }: PostByItemProps) {
...
+                    >
+                        @{profile.handle}
+                    </span>
                 </div>
-            </Fragment>
+                {isSameProfile(currentProfile, profile) ? (
+                    <YesIcon width={40} height={40} className=" relative -right-[10px]" />
+                ) : currentProfile.source === SocialPlatform.Lens ? (
+                    <button
+                        className=" font-bold text-blueBottom"
```
Comment (guanbinrui) [inline src/components/Compose/PostByItem.tsx:73]: Add `disabled` style for this button.

File: src/components/Compose/ComposeSend/useSendFarcaster.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import { useCustomSnackbar } from '@/hooks/useCustomSnackbar.js';
+import { ComposeModalRef } from '@/modals/controls.js';
+import { HubbleSocialMediaProvider } from '@/providers/hubble/SocialMedia.js';
+import { useComposeStateStore } from '@/store/useComposeStore.js';
+import { useFarcasterStateStore } from '@/store/useFarcasterStore.js';
+
+export function useSendFarcaster() {
+    const { type, chars: content } = useComposeStateStore();
+    const enqueueSnackbar = useCustomSnackbar();
+    const currentProfile = useFarcasterStateStore.use.currentProfile?.();
```
Comment (guanbinrui) [inline src/components/Compose/ComposeSend/useSendFarcaster.ts:14]:
```suggestion
    const currentProfile = useFarcasterStateStore.use.currentProfile();
```

File: src/components/Compose/ComposeSend/index.tsx (hunk 5 above)
Comment (UncleBill) [reply src/components/Compose/ComposeSend/index.tsx:21]: Both `sendLens` and `sendFarcaster` have profile checking inside.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**